src/sdk/data-api.ts

```diff
@@ -283,3 +278,57 @@ export class DataApiImpl implements DataApi {
     throw error;
   }
 }
+
+async function buildRecord(conn: Connection, data: any): Promise<Record> {
+  const type = data.attributes.type;
+  const fields = createCaseInsensitiveMap(data);
+  delete fields["attributes"];
+
+  // For any known binaryFields, eagerly fetch the data from the specified
+  // endpoint.
+  const binaryFields = {};
+  if (type in knownBinaryFields) {
+    for (const binFieldName of knownBinaryFields[type]) {
+      if (fields[binFieldName]) {
+        const body: string = await conn.request(fields[binFieldName]);
+        binaryFields[binFieldName] = Buffer.from(body, "binary");
+      }
+    }
+  }
+
+  for (const _ in binaryFields) {
+    return {
+      type,
+      fields,
+      binaryFields: createCaseInsensitiveMap(binaryFields),
+    };
+  }
+  return { type, fields };
+}
+
+function buildUploadFields(record: Record): { [key: string]: unknown } {
+  const fields = {};
+  // Automatically base64 encode any known binaryFields without overwriting existing fields.
+  if (record.type in knownBinaryFields) {
+    for (const binFieldName of knownBinaryFields[record.type]) {
+      if (
+        Buffer.isBuffer(record.binaryFields[binFieldName]) &&
+        (record.fields[binFieldName] === undefined ||
```

Comment:
I wanted to make sure we weren't overwriting any data the user already put there. I don't feel strongly about this though. Should we drop this condition?

Reply:
i'm not sure, i just wanted to understand the reasoning 😅 

let's say i have a `ContentVersion` record that i want to update and i call:

```js
dataApi.update({
  type: 'ContentVersion',
  fields: {
    VersionData: 'this should not be set as a field because it is binary data'
  },
  binaryFields: {
    VersionData: bufferOfBinaryData
  }
})
```

the code, if i'm understanding it correctly, would ignore the `VersionData` value in `binaryFields` and leave the one from `fields`?  if that's true wouldn't that be an error?  or am i just misunderstanding this case?

Reply:
We're now throwing an error in this scenario instead: https://github.com/forcedotcom/sf-fx-runtime-nodejs/pull/400/commits/fb99693d129a9748e85feb3413806ff5b03de7a1